.pre-commit-config.yaml

```diff
@@ -7,15 +7,15 @@ repos:
   -   id: trailing-whitespace
   -   id: end-of-file-fixer
 - repo: https://github.com/psf/black
-  rev: "23.7.0"
+  rev: "23.12.1"
   hooks:
   -   id: black
 - repo: https://github.com/astral-sh/ruff-pre-commit
-  rev: v0.0.282
+  rev: v0.1.3
   hooks:
   - id: ruff
 - repo: https://github.com/RobertCraigie/pyright-python
-  rev: v1.1.320
+  rev: v1.1.348
```

Comment:
I've had issues appear with 1.1.348 which seems unjustified. I reverted to 347. If you're not affected, it's OK ; just keep it in mind.